### DISCUSSION TITLE
Retrieving stac that has equal are to boundary polygon

### DISCUSSION THREAD

**Discussion by nilsleh** [gjoseph92/stackstac#135] [Q&A]
I am looking for a way to to retrieve an image from stackstac that has equal area to the polygon used as boundaries. In practice the polygon is likely to not be perfectly aligned but rotated causing the retrieved image to be closer to the envelope area of the polygon and not the "real" area. So assuming a 1ha area of a polygon, when retrieving an image with stackstac at resolution of 10m, I would want a 10x10pixel image, regardless of the orientation of the polygon. I have made a gist with an [example](https://gist.github.com/nilsleh/50e4b1b409fac8bfc8bb9d7c717c4bfe) and was hoping for an idea of how this can be achieved. Thank you very much in advance!

**Comment by gjoseph92** [comment]
@nilsleh unfortunately, this isn't possible right now with stackstac. It sounds like you'd want to be able to use a non-orthographic geotransform, so that you can make the output images rotated instead of always north-up. This isn't supported right now; see https://github.com/gjoseph92/stackstac/issues/93#issuecomment-982246640 for some discussion.

The easiest thing stackstac could do currently is allow you to pass in a [`RasterSpec`](https://stackstac.readthedocs.io/en/latest/api/internal/stackstac.raster_spec.RasterSpec.html), instead of computing one from the STAC metadata. You'd have to construct the `RasterSpec` yourself (pick bounds and transform, and we'd have to allow `RasterSpec`s to have custom transforms), but if stackstac supported that, I think everything would just work. I would be open to a PR doing that if you wanted to make one.

Otherwise, you'll need to just accept the images in north-up orientation, then apply a function to them after to rotate and clip them (like [`rio.warp.reproject`](https://rasterio.readthedocs.io/en/latest/api/rasterio.warp.html#rasterio.warp.reproject) mapped with [`blockwise`](https://docs.dask.org/en/latest/generated/dask.array.blockwise.html), or with some minor tweaks to take geotransforms into account, [`stackstac.reproject_array`](https://github.com/gjoseph92/stackstac/blob/1c6553a5b6b2acbb89388bd8b1c2d280ea2f3bd5/stackstac/geom_utils.py#L210-L310) would be able to handle this as well).
